File: data/src/main/java/com/github/chojmi/inspirations/data/entity/people/UserEntityImpl.java
package com.github.chojmi.inspirations.data.entity.people;

import android.os.Parcelable;

import com.github.chojmi.inspirations.data.entity.helpers.ContentHolder;
import com.github.chojmi.inspirations.domain.entity.people.UserEntity;
import com.google.auto.value.AutoValue;
import com.google.gson.Gson;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.SerializedName;

@AutoValue
public abstract class UserEntityImpl implements Parcelable, UserEntity {
    public static TypeAdapter<UserEntityImpl> typeAdapter(Gson gson) {
        return new AutoValue_UserEntityImpl.GsonTypeAdapter(gson);
    }

    public abstract String getId();

    @SerializedName("username")
    abstract ContentHolder getUsernameContentHolder();

    @Override
    public String getUsername() {
        return getUsernameContentHolder().getContent();
    }
}

File: data/src/main/java/com/github/chojmi/inspirations/data/entity/people/UserEntityImpl.kt
package com.github.chojmi.inspirations.data.entity.people

import com.github.chojmi.inspirations.data.entity.helpers.ContentHolder
import com.github.chojmi.inspirations.domain.entity.people.UserEntity

data class UserEntityImpl(private val id: String,
                          private val username: ContentHolder) : UserEntity {
    override fun getId(): String = id

    override fun getUsername(): String = username.content
}